mikrotik/provider.go

```diff
@@ -100,6 +97,35 @@ func Provider(client *mt.Mikrotik) *schema.Provider {
 		caCertificate := d.Get("ca_certificate").(string)
 		insecure := d.Get("insecure").(bool)
 
+		if v := os.Getenv("MIKROTIK_HOST"); v != "" {
```

Comment:
What was the motivation behind this change? Is the previous way incompatible with the new plugin framework? It doesn't appear that the `diag.Diagnostics` is used outside of this function so I'm not sure the effect this has.

Reply:
>Is the previous way incompatible with the new plugin framework?

that's right - new `framework` doesn't have `DefaultFunc`, but for `mux` server [provider schema must match](https://developer.hashicorp.com/terraform/plugin/mux/combining-protocol-version-5-providers#requirements).

As for `diags` - nice catch! It should be returned from the function. The fix is on its way

Reply:
Thanks for the clarification and fixing the missing`diag` 👍